### ISSUE TITLE
width goes beyond parent component.

### ISSUE THREAD

**Issue by ameyarumbleon** [indiespirit/react-native-chart-kit#506]
I am trying to add the graph inside my card. I have set the background colors for better understanding.
Red is the container
white is the card component and blue is the view inside the component.
 <LineChart
    data={{
      labels: ["January", "February", "March", "April", "May", "June"],
      datasets: [
        {
          data: [
            Math.random() * 100,
            Math.random() * 100,
            Math.random() * 100,
            Math.random() * 100,
            Math.random() * 100,
            Math.random() * 100
          ],
          color: (opacity = 1) => `rgba(238, 183, 107, ${opacity})`,
        },
        {
          data: [
            Math.random() * 100,
            Math.random() * 100,
            Math.random() * 100,
            Math.random() * 100,
            Math.random() * 100,
            Math.random() * 100
          ]
        }
      ]
    }}
    
    // width={Dimensions.get("window").width} // from react-native
    width= {width}
    height={260}
    yAxisLabel="$"
    yAxisSuffix="k"
    yAxisInterval={1} // optional, defaults to 1
    chartConfig={{
      backgroundColor: "#e26a00",
      backgroundGradientFrom: "#fb8c00",
      backgroundGradientTo: "#ffa726",
      decimalPlaces: 2, // optional, defaults to 2dp
      color: (opacity = 1) => `rgba(255, 255, 255, ${opacity})`,
      labelColor: (opacity = 1) => `rgba(255, 255, 255, ${opacity})`,
      style: {
        borderRadius: 8,
        width:"100%"
      },
      propsForDots: {
        r: "6",
        strokeWidth: "2",
        stroke: "#ffa726"
      }
    }}
    bezier
    style={{
      marginVertical: 8,
      
      borderRadius: 8,
      // width:"100%"
      
    }}
  />
[
![IMG_0156](https://user-images.githubusercontent.com/35232412/117504721-8a6f7b00-af48-11eb-867f-df93f2df9460.jpg)
](url)

**Comment by sajeezy** [comment]
@ameyarumbleon how did you solve this issue?

**Comment by mathcale** [comment]
> @ameyarumbleon how did you solve this issue?

Had the same problem and solved by getting the parent view's width via the `onLayout` prop, setting it on state and using it on chart's `width` prop. Something like this:

```jsx
const [chartParentWidth, setChartParentWidth] = useState(0);

<View
	onLayout={({ nativeEvent }) => setChartParentWidth(nativeEvent.layout.width)}
	style={styles.chartWrapper}
>
	<LineChart
		...props
		width={chartParentWidth}
	/>
</View>
```

**Comment by MadelinFg** [comment]
> > @ameyarumbleon how did you solve this issue?
> 
> Had the same problem and solved by getting the parent view's width via the `onLayout` prop, setting it on state and using it on chart's `width` prop. Something like this:
> 
> ```js
> const [chartParentWidth, setChartParentWidth] = useState(0);
> 
> <View
> 	onLayout={({ nativeEvent }) => setChartParentWidth(nativeEvent.layout.width)}
> 	style={styles.chartWrapper}
> >
> 	<LineChart
> 		...props
> 		width={chartParentWidth}
> 	/>
> </View>
> ```

Is not working for me, i'm getting 
`Invariant Violation: [3967,"RCTView",31,{"onStartShouldSetResponder":true,"onMoveShouldSetResponder":true,"onResponderGrant":true,"onResponderMove":true,"onResponderEnd":true,"onResponderTerminationRequest":true,"accessibilityState":{},"accessibilityValue":{},"position":"absolute","height":293,"bottom":57,"width":"<<Infinity>>","zIndex":0}] is not usable as a native method argument`

using it like this
`<View onLayout={({ nativeEvent }) => setChartParentWidth(nativeEvent.layout.width)}>
<LineChart 
...props
width={userControls.chartParentWidth}
adjustToWidth />
</View>`

**Comment by adam-kuhn** [comment]
> > [@ameyarumbleon](https://github.com/ameyarumbleon) how did you solve this issue?
> 
> Had the same problem and solved by getting the parent view's width via the `onLayout` prop, setting it on state and using it on chart's `width` prop. Something like this:
> 
> const [chartParentWidth, setChartParentWidth] = useState(0);
> 
> <View
> 	onLayout={({ nativeEvent }) => setChartParentWidth(nativeEvent.layout.width)}
> 	style={styles.chartWrapper}
> >
> 	<LineChart
> 		...props
> 		width={chartParentWidth}
> 	/>
> </View>


This was a big help, but only got me part of the way there. For some reason I still needed to offset the y-axis by 10. The default label offset is 35 (https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/blob/master/docs/LineChart/LineChartProps.md#onfocus-and-strip-related-props:~:text=1-,yAxisLabelWidth,-number)
, and I needed to subrtract another 10 to make the chart fit properly. 🤷‍♂ 

```
          <View
            style={{
              flex: 1,
              width: "90%",
            }}
            onLayout={({ nativeEvent }) =>
              setChartParentWidth(nativeEvent.layout.width)
            }
          >
            <LineChart
              width={chartParentWidth - 45}
              yAxisLabelWidth={35}
           .....
           />
         </View>

**Comment by mwannags** [comment]
I am having similar issues here!

I am trying to make the chart reponsive related to its parent <View> but it seems like nothing "clean" works...
I did not set the width explicitly, so it will default to its parent – but it doesn't. So I will try out the onLayout stuff, but it's not really the solution I'd wish for.

Maybe someone has a better solution? I've looked at the victory-charts but they do not offer the same amount of features as the gifted chars
